tools/ctl/ctl.c

```diff
@@ -508,6 +509,8 @@ static int ctl_set_get(struct ctl_data *ctl_data)
 		fprintf(stdout, "Retrieving configuration for ");
 		fprintf(stdout, "device %s control %s.\n",
 			ctl_data->dev, ctl_data->cname);
+		/* set the ABI header to pass the param ID */
+		header_init(ctl_data);
```

Comment:
Byte control set/get has been working without this change, I just checked. I wonder if Linux kernel has set this so missing this has had no impact.

Reply:
It works because the driver always sends the cached data to the userspace. If we had volatile controls that need to read the data from the DSP, this would be needed